### ISSUE TITLE
Organize Communication

### ISSUE THREAD

**Issue by DragonmasterJ** [brata-hsdc/brata#85]
Need to decide standard communication tools to use (forum, chat client, pull requests) that can persist conversations, keep discussions private if necessary (i.e. away from students), and makes it easier to collaborate when working simultaneously when separate (i.e. at home).

**Comment by DragonmasterJ** [comment]
Using issues for discussion sounds like an interesting idea, as long as we are discussing things that can be left public, though it begs the question whether we have a couple issues always open for general discussion, etc. The only issue if if we have to migrate away from GitHub, will the next service import our issues.

**Comment by ellerychan** [comment]
When I reply to this list, I'm not really sure who I'm replying to.  I
hope I'm not generating a lot of noise for people who don't care.

I kind of like discussing within the issue thread.  It keeps the
topic-specific discussion all together.  I agree the public/private
visibility is a concern.

-ellery-

**Comment by ellerychan** [comment]
Can you figure out a way to attach all the Issues labeled `Component-MS` to the **brata-hsdc/brata.masterserver** repo?

**Comment by jawaad-ahmad** [comment]
I'm not getting any e-mails about these posts as far as I can tell if you're worried about noise. Maybe it depends on who's watching the issue if such a feature is available here.

There are a couple  out there that seem to have interesting features. Here are some reviews:

http://forum-services-review.toptenreviews.com/

https://forums.digitalpoint.com/threads/what-is-the-best-free-forum-hosting-website.2661573/

One thing to keep in mind is which ones let us control who gets to have an account, maybe nothing that lets a user self-register in order to protect against spam.

I'm liking our issue discussions right now as well, so conversations within issues seems like a good plan for now.

The exception will be when we need conversations and discussions outside of an issue. If we continue with this, then do we open an issue just to have the discussion and then have the issue closed? Maybe that's acceptable.

**Comment by ellerychan** [comment]
I would vote for keeping the discussion here.  Even with minor shenanigans, it seems better than adding another location that is essentially disconnected from this activity.

This seems to be working.

**Comment by sminster** [comment]
@jawaad-ahmad you're not getting emails?  I'm getting plenty.  Though I don't mind, I prefer the push notification.  But maybe there is a setting I have that you don't?

**Comment by jawaad-ahmad** [comment]
It looks like I was ignoring a couple repositories in my profile. I set them to **Stop Ignoring**. Maybe that will help. Then again, depending on the volume of e-mails maybe that will hurt!

Very well then. If we keep discussions here, then I understand that to mean we are free to open issues and use up issue numbers in order to ask questions by marking the issue with a `question` label as described on the [Process](../../brata.masterserver/wiki/Process) page.

I have no problem with that, but I want it to be clear if we go down this route that the culture will include opening issues for discussions that don't involve code changes, and that we are OK with this.

I'm moving this to `in review` on behalf of @DragonmasterJ.

**Comment by DragonmasterJ** [comment]
I think Gists are more powerful than I first imagined, for discussions that don't have a place in issues or within the pull requests I think we will have a lot of room to take advantage of them for discussion. It looks like each Gist discussion(?) is treated like a git repository and we can make them public or private. Will be looking into this over the weekend.

@jawaad-ahmad @ellerychan Are there any discussions we've been having in the MS pull requests or in the issues that feel disjointed that we may want to put up in a Gist to try it out? If not I'll just start one to try it out.

**Comment by jawaad-ahmad** [comment]
Nothing comes to mind. Just keep in mind gists are very public and searchable and browsable. We can't have any conversations about things such as Mike's ciphers.

**Comment by DragonmasterJ** [comment]
Yeah I just realized before I looked into it that if it said it's just treated as a another git repo, private only limits the contributions, not the visibility. Private information still looks like we may just have to discuss outside of GitHub.

**Comment by jawaad-ahmad** [comment]
We don't have that much private conversation. For that little bit, we can
stick to phone, face-to-face, and our regular e-mail.

For instance, the encryption cipher documents are all on the internal
SharePoint; the implementation is straightforward enough that it can be
written once and then e-mailed or even walked over to Mike.

By the way, I'm replying to some of these such as this post via e-mail
rather than logging onto the site. Looking at the posts created from these
replies, it looks pretty good. So we can have some of these conversations
via e-mail replies and they all will make it to the GitHub site for others
to view, browse, search, and implement when the time comes.
